x-pack/plugins/apm/public/components/shared/kuery_bar/index.tsx

```diff
@@ -95,22 +102,22 @@ export function KueryBar(props: { prepend?: React.ReactNode | string }) {
         (await data.autocomplete.getQuerySuggestions({
           language: 'kuery',
           indexPatterns: [indexPattern],
-          boolFilter: getBoolFilter({
-            groupId,
-            processorEvent,
-            serviceName,
-            environment,
-            urlParams,
-          }),
+          boolFilter:
+            props.boolFilter ??
+            getBoolFilter({
+              groupId,
+              processorEvent,
+              serviceName,
+              environment,
+              urlParams,
+            }),
           query: inputValue,
           selectionStart,
           selectionEnd: selectionStart,
           useTimeRange: true,
           method: 'terms_agg',
         })) || []
-      )
-        .filter((suggestion) => !startsWith(suggestion.text, 'span.'))
```

Comment:
Why was this removed? We don't want to display span fields afair

Reply:
`span.destination.service.resource` certainly is valuable. Do you think it still makes sense to filter out other `span.*` fields given the amount of noise (there are hundreds of fields that will probably never be used to search on)?

Reply:
Let me know if you still want this reverted, I'll open an issue or PR if appropriate.

Reply:
It's fine.


Just to make sure I understand this correctly: 

> Do you think it still makes sense to filter out other span.* fields given the amount of noise (there are hundreds of fields that will probably never be used to search on)?

You are saying that there's already a lot of noise, so the additional span fields won't matter? That's probably true but also sad. Would be good if we could exclude the fields that don't exist in the selected dataset.

Reply:
Yeah, agreed, I think we're tracking that in https://github.com/elastic/kibana/issues/94879